es6-shim.js

```diff
@@ -2097,6 +2097,88 @@
     // Shim incomplete iterator implementations.
     addIterator(Object.getPrototypeOf((new globals.Map()).keys()));
     addIterator(Object.getPrototypeOf((new globals.Set()).keys()));
+
+    // Reflect
+    if (!globals.Reflect) {
+      globals.Reflect = {
+
+        // Syntax in a functional form.
+        get: function (target, key, receiver) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          return target[key];
+        },
+
+        set: function (target, key, value, receiver) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          target[key] = value;
+        },
+
+        has: function (target, key) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          return key in target;
+        },
+
+        deleteProperty: function (target, key) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          delete target[key];
+        },
+
+        enumerate: function (target) {
+          if (!ES.TypeIsObject(target)) {
+            throw new TypeError('target must be an object');
+          }
+
+          var keys = [];
+
+          for (var key in target) {
+            keys.push(key);
+          }
+
+          return new ArrayIterator(keys, 'value');
```

Comment:
I tested @zloirock's enumerate implementation in Chome and Firefox, and it seems like the possible keys which the iterator can yield are determined when `iter.next()` is first called. Properties defined afterwards are not included, nor any deleted properties.

```javascript
var obj = { a: 1, b: 2 }, iter = Reflect.enumerate(obj);
iter.next(); // { value: "a", done: false }

obj.c = 3;
iter.next(); // { value: "b", done: false }
iter.next(); // { value: undefined, done: true }
```
```javascript
var obj = { a: 1, b: 2 }, iter = Reflect.enumerate(obj);

obj.c = 3;
iter.next(); // { value: "a", done: false }
iter.next(); // { value: "b", done: false }
iter.next(); // { value: "c", done: false }
iter.next(); // { value: undefined, done: true }

```
```javascript
var obj = { a: 1, b: 2 }, iter = Reflect.enumerate(obj);

iter.next(); // { value: "a", done: false }
delete obj.b;
iter.next(); // { value: undefined, done: true }

```